### ISSUE TITLE
Provide possibility to reuse Builder and Reader

### ISSUE THREAD

**Issue by mzaks** [mzaks/FlatBuffersSwift#27]
As discussed in #25

**Comment by hassila** [comment]
I think this is one of the biggest wins for repeated building&reading - e.g. the builder has a bunch of properties that are objects and needs to be set up / allocated regardless of whether cache is in use - e.g. 

```
    public var cache : [ObjectIdentifier : Offset] = [:]
    public var inProgress : Set<ObjectIdentifier> = []
    public var deferedBindings : [(object:Any, cursor:Int)] = []

```

Would be nice to add something like

```
    public static func fromReader(r : FlatBufferReader) -> FooBarContainer {
        let reader = r
        let objectOffset = reader.rootObjectOffset
        return create(reader, objectOffset : objectOffset)!
    }

and 

    public func toByteArray (b :FlatBufferBuilder) -> [UInt8] {
        let builder = b
        let offset = addToByteArray(builder)
        performLateBindings(builder)
        return try! builder.finish(offset, fileIdentifier: nil)
    }

```

and corresponding reader&builder 'reset' methods.

**Comment by hassila** [comment]
A lazy instance would probably want to keep its reader-per-instance though (can't think of something decent there right now otherwise).

**Comment by hassila** [comment]
So just to keep track after #37 was merged to handle the builder path, codegen needs to be tweaked to:

```
public extension FooBarContainer {
    public func toByteArray (config : BinaryBuildConfig = BinaryBuildConfig()) -> [UInt8] {
-->        let builder = FlatBufferBuilder.create(config)
        let offset = addToByteArray(builder)
        performLateBindings(builder)
-->        let result = try! builder.finish(offset, fileIdentifier: nil)
-->        FlatBufferBuilder.reuse(builder)
-->        return result
    }
}


```

**Comment by hassila** [comment]
Addressed second half in #40 - I think we can close this one after that one is integrated.
